app/components/file/browse/blob-files-browser/blob-files-browser.component.ts

```diff
@@ -52,11 +57,40 @@ export class BlobFilesBrowserComponent implements OnChanges, OnDestroy {
                 this.upload(event).subscribe(() => {
                     this.fileNavigator.refresh(path);
                 });
+
                 return Observable.of(null);
             },
         });
     }
 
+    public handleDeleteEvent(event: FileDeleteEvent) {
+        const { path } = event;
+        console.log("BlobFilesBrowserComponent.handleDeleteEvent: ", event);
+
+        // TODO: Check with Tim as to whether i can chain 2 background tasks to run one after another.
+        // In which case most of this can be moved into DataContainerFilesComponent.
+        // task 1: get files to delete
+        // task 2: delete files
+
+        this.dialogService.confirm(`Delete files`, {
+            description: event.isDirectory
+                ? `All files will be deleted from the folder: ${path}`
+                : `The file '${FileUrlUtils.getFileName(path)}' will be deleted.`,
+            yes: () => {
+                const listParams = { recursive: true, startswith: path };
+                const data = this.storageService.listBlobs(Promise.resolve(this.container), listParams);
+                return data.fetchAll().flatMap(() => data.items.take(1)).map((items) => {
```

Comment:
i would not use .map here because it won't run unless you subscribe to it. I know this funciton is being passed up to the dialog which subscribe to it but it is a bit confusing to actually do state actions in the map. I would just have it 
const obs = ...
obs.subscribe(...)
return obs;